compiler/p/codegen/ControlFlowEvaluator.cpp

```diff
@@ -1252,6 +1254,12 @@ TR::Register *OMR::Power::TreeEvaluator::compareIntsForEquality(TR::InstOpCode::
       site->setNode(node);
       site->setAconstNode(secondChild);
       }
+   else
+      {
+      TR_ASSERT(!(node->isNopableInlineGuard()),"We should not reach this case with nopable guard node");
```

Comment:
I suggest we use a more explicit assert. The message should describe _why_ we shouldn't reach this case with a nopable guard. At the moment it provides no feedback to the user should the assertion trigger. Perhaps saying:

```
TR_ASSERT(!(node->isNopableInlineGuard()), "NOPable guards register runtime assumptions and do not need relocation records, hence class or method pointer constants underneath NOPable guards should not be evaluated");
```

Similarly in other places in this commit.